packages/dd-trace/src/profiling/profilers/cpu.js

```diff
@@ -0,0 +1,103 @@
+'use strict'
+
+const dc = require('diagnostics_channel')
+
+const beforeCh = dc.channel('dd-trace:AsyncResourceStorage:before')
+const afterCh = dc.channel('dd-trace:AsyncResourceStorage:after')
+
+function getActiveSpan () {
+  const scope = global._ddtrace.scope()
+  if (!scope) return
+  return scope.active()
+}
+
+function getStartedSpans (activeSpan) {
+  const context = activeSpan.context()
+  if (!context) return
+  return context._trace.started
+}
+
+function getSpanContextTags (span) {
+  return span._context()._tags
+}
+
+function isWebServerSpan (tags) {
+  return tags['span.type'] === 'web' && tags['span.kind'] === 'server'
```

Comment:
`tags['span.type'] === 'web'` is always a server.

Reply:
A client request is something else?

Reply:
Client requests can be `http` or some other values for other transports, but `web` always means HTTP server.